content/geometry/PolygonUnionArea.h

```diff
@@ -0,0 +1,49 @@
+/**
+ * Author: black_horse2014, chilli
+ * Date: 2019-10-29
+ * License: CC0
+ * Source:
+ * Description: Calculates the area of the union of $n$ polygons (not necessarily convex).
+ * Status: Submitted on ECNA 2017 Problem A.
+ * Time: $O(N^2)$, where $N$ is the total number of points
+ * Usage:
+ */
+#pragma once
+
+#include "Point.h"
+#include "sideOf.h"
+
+typedef Point<double> P;
+double rat2(P a, P b) { return sgn(b.x) ? a.x/b.x : a.y/b.y; }
```

Comment:
(this function is super confusing to me btw)

Reply:
Honestly, I don't really understand what this function does either... It has to do with the case when  2 line segments have infinite intersection points. Kind of icky.